### ISSUE TITLE
Changelog URLs not always generated correctly when using shorthand or git: syntax

### ISSUE THREAD

**Issue by pdehaan** [ajoslin/conventional-changelog#49]
In our package.json, we're using the shorthand format for the `repository` field, "mozilla/fxa-content-server". We noticed that when generating the changelog (via [grunt-conventional-changelog](https://github.com/btford/grunt-conventional-changelog)) that the links in the CHANGELOG.md don't resolve properly:

Using shorthand in the package.json:
```js
"repository": "mozilla/fxa-content-server",
```

Generates this:
```markdown
<a name="0.33.0"></a>
## 0.33.0 (2015-03-23)


#### Bug Fixes

* **test:** Fix the account locked tests when run against latest. ([45073cf6](mozilla/fxa-content-server/commit/45073cf66cf69118251c85846f6ce0e330f8bafd))


#### Features

* **client:** Add the account locked flows. ([8e405298](mozilla/fxa-content-server/commit/8e405298b8fee3b1e5c4e495a006d42580432c49))
```

---

If we switch to the long version of `repository`:
```js
  "repository": {
    "type": "git",
    "url": "https://github.com/mozilla/fxa-content-server"
  },
```

We get this (fully qualified URLs):
```markdown
<a name="0.33.0"></a>
## 0.33.0 (2015-03-23)


#### Bug Fixes

* **test:** Fix the account locked tests when run against latest. ([45073cf6](https://github.com/mozilla/fxa-content-server/commit/45073cf66cf69118251c85846f6ce0e330f8bafd))


#### Features

* **client:** Add the account locked flows. ([8e405298](https://github.com/mozilla/fxa-content-server/commit/8e405298b8fee3b1e5c4e495a006d42580432c49))
```

---

Similarly, if you use a repository url of git:// then it isn't convertingand parsing correctly:

```js
  "repository": {
    "type": "git",
    "url": "git@github.com:mozilla/fxa-content-server.git"
  },
```

Then it generates links like this:
```markdown
<a name="0.33.0"></a>
## 0.33.0 (2015-03-23)


#### Bug Fixes

* **test:** Fix the account locked tests when run against latest. ([45073cf6](git@github.com:mozilla/fxa-content-server/commit/45073cf66cf69118251c85846f6ce0e330f8bafd))


#### Features

* **client:** Add the account locked flows. ([8e405298](git@github.com:mozilla/fxa-content-server/commit/8e405298b8fee3b1e5c4e495a006d42580432c49))
```

Long story longer, it doesn't look like we're normalizing the Git URLs everywhere and assuming that the repository URL is HTTP-esque.

**Comment by stevemao** [comment]
What version are you using? Conventional-changelog already fixed this issue and publish. Grunt-conventional-changelog is waiting on @brford cos he's traveling. Can you try using the master branch and see if it works for you? Thank you.

**Comment by pdehaan** [comment]
Sorry, I was using **grunt-conventional-changelog@1.1.0**, which uses **conventional-changelog@0.0.16**.

**Comment by stevemao** [comment]
grunt-conventional-changelog@1.1.0 is not using conventional-changelog@0.0.16, can you try the master branch of grunt-conventional-changelog?

**Comment by pdehaan** [comment]
Yeah, it seems to be working if I test w/ master:
```js
"grunt-conventional-changelog": "git://github.com/btford/grunt-conventional-changelog",
```

**Comment by stevemao** [comment]
Thanks. We will publish it once we can.

**Comment by stevemao** [comment]
For the reference, this was fixed in cb1feb7dd8f14cc55472fd0fbc229f23782c7582
